File: Structural/Bridge/Python/tv.py
# tv.py

from device import Device

class TV(Device):
  def __init__(self):
    self._on = False
    self._volume = 50

  def turn_on(self):
    self._on = True
    print("TV is now ON.")

  def turn_off(self):
    self._on = False
    print("TV is now OFF.")

  def set_volume(self, volume: int):
    self._volume = volume
    print(f"TV volume set to {volume}")

  def get_volume(self) -> int:
    return self._volume

  def is_on(self) -> bool:
    return self._on

File: Structural/Bridge/JavaScript/tv.js
// tv.js

const Device = require("./device");

class TV extends Device {
  constructor() {
    super();
    this.on = false;
    this.volume = 50;
  }

  turnOn() {
    this.on = true;
    console.log("TV is now ON.");
  }

  turnOff() {
    this.on = false;
    console.log("TV is now OFF.");
  }

  setVolume(volume) {
    this.volume = volume;
    console.log(`TV volume set to ${volume}`);
  }

  getVolume() {
    return this.volume;
  }

  isOn() {
    return this.on;
  }
}

module.exports = TV;
